src/firefly/js/visualize/ChartsTableViewPanel.jsx

```diff
@@ -130,33 +139,28 @@ class ChartsPanel extends React.Component {
     }
 
     componentWillUnmount() {
-        const {tblId} = this.props;
-        const popupId = getChartOptionsPopupId(tblId);
         if (isDialogVisible(popupId)) {
-            hideChartOptionsPopup(tblId);
+            hideChartOptionsPopup();
         }
     }
 
     handlePopups() {
-        const {tblId, optionsPopup} = this.props;
-        const popupId = getChartOptionsPopupId(tblId);
-        if (optionsPopup) {
+        if (this.props.optionsPopup) {
             const {optionsShown, chartType} = this.state;
-            const {tableModel, tblStatsData} = this.props;
             if (optionsShown) {
+                const {tableModel, tblStatsData, tblId, chartId} = this.props;
                 // show options popup
-                const popupId = getChartOptionsPopupId(tblId);
                 let popupTitle = 'Chart Options';
 
                 const reqTitle = get(tableModel, 'tableMeta.title');
-                if (reqTitle) { popupTitle += ': '+decodeURIComponent(reqTitle); }
+                if (reqTitle) { popupTitle += ': '+chartId+' '+decodeURIComponent(reqTitle); }
 
                 var popup = (
                     <PopupPanel title={popupTitle} closeCallback={()=>{this.toggleOptions();}}>
                         <div
                             style={{overflow:'auto',width:OPTIONS_WIDTH,height:300,paddingTop:10,paddingLeft:10,verticalAlign:'top'}}>
-                            {this.renderChartSelection()}
-                            <OptionsWrapper {...{tblId, tableModel, tblStatsData, chartType}}/>
+                            {chartId===tblId ? this.renderChartSelection() : false}
```

Comment:
It's only for the current firefly app scenario: one chart per table. In future this scenario might go away, but I was not sure I did not want to break it now. It's special, because in this case we display a choice between Scatter Plot and Histogram, which we don't give if the chart type is known (api case).